File: regional/regional/2015/I_O/Classy/Classy.pl
my %remap = ('middle'=>2, 'upper'=>1, 'lower'=>3) ;
scalar <> ;
while (<>) {
   chomp ;
   s/ class// ;
   s/:// ;
   @f = split " ", $_ ;
   $name = shift @f ;
   $key = join '', map { $remap{$_} } reverse @f ;
   $key .= '2' while length($key) < 100 ;
   push @a, "$key $name" ;
}
print((split " ")[1], "\n") for sort @a ;

File: regional/PacificNW/2015/I_O/Classy/Classy.py
import sys
who = []
d = dict(upper='1',middle='2',lower='3')
raw_input()
while True:
   try:
      lin = raw_input()
      clas = lin.split(":")[1].split(" ")[::-1]
      key = ''.join(map(lambda s:d[s],clas[1:-1]))
      key = (key + ('2' * 255))[:255] ;
      who.append(key + ":" + lin)
   except (EOFError):
      break
who = sorted(who)
for lin in who:
   print lin.split(":")[1]
